### PR TITLE
Week 1

### PR DESCRIPTION
**Pull request by dlfivefifty**

### CODE REVIEW

File: src/labs/lab1s.jl [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+# Numerical analysis primarily studies the mathematical construction and analysis of algorithms
+# for approximating solutions to continuous problems like computing integrals or solving differential equations.
+# It is fundamental to the area to also understand how to implement such algorithms
+# in software. In year 1 you learned basic programming concepts such as loops, conditions,
+# functions, etc. and in this first lab we will employ these concepts in the implementation
+# of some basic approximations you have already seen in the guise of analysis.
+# In particular, we will look at the following:
+
+# 1. Integeration via the rectangular rule:  integrals are defined by
+# approximating a function by rectangles and taking the limit as the number of rectangles becomes
```
Comment (dlfivefifty) [inline src/labs/lab1s.jl]: In my viewer this is rendered on multiple lines

File: src/labs/lab1s.jl [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+## END
+@test trapeziumrule(exp, 1000) ≈ exp(1) - 1 atol=1E-6
+
+# **Exercise 1(d)** Compare `rightrectangularrule` and `trapeziumrule`
+# for $f(x)$ equal to $\cos(2πx)$ and $\sin(4πx)$ for $n = 1, 3, 5$. Mathematically,
+# should they return the same result? Do they in practice? Does the accuracy improve if we make `n = 10_000`?
+# Hint: It is possible to type `π` by typing `\pi` and then hitting the tab key. Otherwise one can just
+# type out `pi`.
+
+## TODO: test the three functions with varying `n`
```
Comment (dlfivefifty) [inline src/labs/lab1s.jl]:
```suggestion
## TODO: test the three functions defined above with varying `n`
```

**Comment by dlfivefifty** [comment]
@SamBrzezicki I've added the notes for Week 1 too

File: notes/I.1.RectangularRule.tex [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+\]
+In these notes we consider the "Analysis" part of "Numerical Analysis": we want to \emph{prove} the convergence rate of the approximation, including finding an explicit expression for the constant $C$.
+
+To tackle this question we consider the error incurred on a single "rectangle", then sum up the errors on rectangles.
+
+Now for a secret. There are only so many tools available in analysis (especially at this stage of your career), and  one can make a safe bet that the right tool in any analysis proof is either (1) integration-by-parts, (2) geometric series or (3) Taylor series. In this case we use (1):
+
+\begin{lemma}[Rectangular Rule error on one panel] Assuming $f$ is differentiable we have
+\[
+\ensuremath{\int}_a^b f(x) {\rm d}x = (b-a) f(a) +  \ensuremath{\delta}
```
Comment (SamBrzezicki) [inline notes/I.1.RectangularRule.tex]: $(b-a)f(b) +\delta$

**Pull request merged**